### PR TITLE
feat: add templating support

### PR DESCRIPTION
**Pull request by erezrokah**
Hi 👋 

It's me again, from this PR https://github.com/BetaHuhn/repo-file-sync-action/pull/200

Opening as draft to get some feedback.

### CODE REVIEW

File: src/git.js [hunk 1]
```diff
@@ -123,7 +123,7 @@ class Git {
 	async createPrBranch() {
 		const prefix = BRANCH_PREFIX.replace('SOURCE_REPO_NAME', GITHUB_REPOSITORY.split('/')[1])
 
-		let newBranch = path.join(prefix, this.repo.branch).replace(/\\/g, '/')
+		let newBranch = path.join(prefix, this.repo.branch).replace(/\\/g, '/').replace(/\/\./g, '/')
```
Comment (erezrokah) [inline src/git.js:126]: Fix from https://github.com/BetaHuhn/repo-file-sync-action/pull/200

File: src/helpers.js [hunk 2]
```diff
@@ -63,21 +63,29 @@ const pathIsDirectory = async (path) => {
...
+		// don't copy template values files, or templated generated files that don't match the target repo name
+		if (templatingEnabled && (isTemplateValuesFile(file) || isNonMatchingTemplateGeneratedFile(file, repoName))) {
+			return false
+		}
+
 		return true
 	}
 
-	await fs.copy(src, dest, exclude !== undefined && { filter: filterFunc })
+	await fs.copy(src, dest, { filter: filterFunc })
```
Comment (erezrokah) [inline src/helpers.js:88]: We check for `exclude !== undefined` inside  `filterFunc`

**Comment by BetaHuhn** [comment]
Hey @erezrokah,

first of all, thanks for taking the time to contribute to this project!

Next time, could you explain what your changes are trying to achieve in the PR body? This makes it easier for me as a maintainer to get an overview without me having to go through your changes one by one and having to piece everything together myself.

I think templating support could be useful and I would like to include it in this action, but I am not sure if your current syntax is the right way to go about it. Specifically, I don't like the way you need to specify files to use as data for the templates: `<template-name>.<repo>.values.yml`.

I created a [discussion](https://github.com/BetaHuhn/repo-file-sync-action/discussions/204) about this topic where we can discuss how we implement this feature before moving to the actual code. I want to hear about more use cases before finalizing a syntax.

Happy to revisit your idea if you explain your reasoning a bit more. Feel free to comment your ideas in the discussion!

Thanks :)

**Comment by erezrokah** [comment]
Thank you for commenting @BetaHuhn and I appreciate the time you spent going through the PR. I'll definitely add more details next time 🤦  Happy to close the PR and continue this in the discussion 👍

**Pull request closed without merging**